lib/editor/components/ScheduleExceptionForm.js

```diff
@@ -122,12 +122,30 @@ export default class ScheduleExceptionForm extends Component<Props> {
     } = activeEntity
 
     // getRangesForDates requires a sorted array. We also sort the zeroes to the end of the array for display purposes.
+    let addDateDisabled = false // Disable adding a new date if we come across any current validation issues.
     const {ranges, parsedDates} = getRangesForDates([...dates].sort((a: any, b: any) => {
       if (a === 0) return 1
+      else if (b === 0) return -1
       return (a: number) - (b: number)
     })).reduce((acc, cur) => {
-      if (cur.startDate === cur.endDate) acc.parsedDates.push(cur.startDate)
-      else acc.ranges.push(cur)
+      let validationState
+      const dateIndex = dates.findIndex(d => d === cur.startDate)
+      if (cur.startDate === cur.endDate) {
+        validationState = this._checkValidation(`dates-${dateIndex}`)
+        acc.parsedDates.push({
+          date: cur.startDate,
+          validationState
+        })
+      } else {
+        const endIndex = dates.findIndex(d => d === cur.endDate)
+        validationState = this._checkRangeValidation(dateIndex, endIndex)
+        acc.ranges.push({
+          ...cur,
+          validationState
+        })
+      }
+
+      if (validationState === 'error') addDateDisabled = true
```

Comment:
Writing to global variables from inside a reducer is really bad practice! Good news is that the variable isn't needed so this can be removed

Reply:
Yep, this is removed in https://github.com/ibi-group/datatools-ui/pull/843/commits/bcfe4911d3d622f0ebf397e12c47e6b12df3ea45